### DISCUSSION TITLE
I expected SaveAsync(forceUpdate: true) to always hit the Data Portal

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#3241] [Questions] [answered]
If we call `SaveAsync(forceUpdate: true)` on an object that is **NOT** `base.IsNew` and **NOT** `IsDirty` nothing happens.

I've probably misunderstood what this should do, but I was expecting `forceUpdate: true` to force the save to hit the Data Portal, i.e. the same as:
```cs
obj.MarkDirty(); // Can't do this as it's protected

obj = obj.SaveAsync();
```

What's the _proper_ way to force a save on an object which is not dirty, given `MarkDirty()` is `protected`?

**Comment by jhuerta3** [comment]
maybe 

`base.MarkDirty()`

I did something similar but with Delete. I had to set the IsDelete flag to true so that I can run my business rules. 

```
        [Delete]
        private void Delete(Guid equipmentID, [Inject] DataAccess.IEquipmentDal dal)
        {
            base.MarkDeleted();
            EquipmentID = equipmentID;
            MarkForDelete = base.IsDeleted;
            BusinessRules.CheckRules(EquipmentIDProperty);
            if (IsValid)
            {
                dal.Delete(EquipmentID);
            }
        }
```

**Reply by michaelcsikos** [reply]
Yes, I could make a new method in my own base class to call `MarkDirty()` before calling `Save()`. Or I could just "touch" any property. It just seems strange if there's no built-in way to do it.

**Comment by Chicagoan2016** [comment, marked as answer]
If memory serves, forceupdate is for Web stateless applications when we want to run the 'UPDATE' operation in Database, instead of 'INSERT'.

**Reply by rockfordlhotka** [reply]
That's exactly right. the forceupdate option exists to tell the data portal that the operation that _looks like an insert_ is actually an update. Its purpose is to support MVC and similar UI technologies where a postback occurs and you get the data from a remote UI.

**Comment by michaelcsikos** [comment]
So I should just add my own `MarkDirtyAndSaveAsync()` as there's no way for the UI to do this currently without touching a property.

**Reply by rockfordlhotka** [reply]
How are you creating and populating an object such that it isn't new or dirty?

**Reply by michaelcsikos** [reply]
Hi @rockfordlhotka, this is an usual scenario where an object can be saved to two different locations: either in a drawn component in AutoCAD, or saved as a template which writes the object to a JSON file to disk. The user can double-click on a drawn component to open the form and could immediately click **Save as template**. We clone the object, which at this point is not new or dirty, and call save. So we just have to call `MarkDirty()` first so that the save isn't short-circuited by CSLA. There's no `public` way to do this out of the box, though.

**Reply by rockfordlhotka** [reply]
I see. Is Clone virtual, so you could mark the clone as dirty?
